### DISCUSSION TITLE
Does UPnP forward all necessary ports?

### DISCUSSION THREAD

**Discussion by ghost** [nicotine-plus/nicotine-plus#2254] [Q&A] [answered]
I've found an issue with my router where ports I open manually aren't actually being opened but ports forwarded by UPnP are opened properly, so I need to use UPnP.

Does nicotine forward all necessary ports? Because it says that the listening port is 2234, but it says that other ports are also necessary, ports 2235, 2236, 2237, 2238, 2239. So will it forward all of those or only one?

**Comment by slook** [comment, marked as answer]
Only one port is actually nessassry. The first port in the range that isn't already in use by another application or service is selected.
